src/stores/WidgetStore.ts

```diff
@@ -38,6 +38,10 @@ export interface IApp extends IWidget {
     avatar_url?: string; // MSC2765 https://github.com/matrix-org/matrix-doc/pull/2765
 }
 
+export function isAppWidget(widget: IWidget | IApp): widget is IApp {
+    return typeof widget["roomId"] === "string";
+}
+
```

Comment:
In a piece of work I started a `Typeguards.ts` file in the src folder. I'm still not entirely sure if it would be better to put typeguards in one file, or always keep them closer to what they're guarding for, like here.

What do you think? Better to keep here or consolidate in a single typeguards file?

Reply:
My personal preference is closer to the object, in you case I think it makes sense for generic ones which don't apply to a specific type, like `null`